sdk/core/azure-core-experimental/src/main/java/com/azure/core/experimental/util/BinaryData.java

```diff
@@ -185,6 +187,25 @@ public static BinaryData fromBytes(byte[] data) {
         return new BinaryData(data);
     }
 
+    /**
+     * Serialize the given {@link Object} into {@link BinaryData} using the provided {@link JsonSerializer}. This will
+     * require the client to configure Json serializer in classpath.
+     *
+     * @param data The {@link Object} which needs to be serialized into bytes.
+     * @throws NullPointerException if {@code data} is null.
+     * @throws IllegalStateException If cannot find any JSON serializer provider on the classpath.
+     * @return {@link BinaryData} representing binary data. Or {@code null} if it fails to serialize the data.
+     *
+     * @see JsonSerializer
+     */
+    public static BinaryData fromObject(Object data) {
+        Objects.requireNonNull(data, "'data' cannot be null.");
```

Comment:
In the case of a null object do we want to throw or return an empty `BinaryData`? Do we want to force the caller to handle null checking or return them an object which represents no data.